File: LeetCodeCPP/724.find-pivot-index.cpp
/*
 * @lc app=leetcode id=724 lang=cpp
 *
 * [724] Find Pivot Index
 */

#include <vector>

using namespace std;

// @lc code=start
class Solution
{
public:
    int pivotIndex(vector<int> &nums)
    {
        int sum = 0, leftSum = 0;
        for (auto num : nums)
            sum += num;

        for (size_t i = 0; i < nums.size(); i++)
        {
            if (leftSum == sum - leftSum - nums[i])
                return i;
            leftSum += nums[i];
        }
        return -1;
    }
};
// @lc code=end

File: LeetCodePython/724.find-pivot-index.py
#
# @lc app=leetcode id=724 lang=python3
#
# [724] Find Pivot Index
#

from typing import List


# @lc code=start
class Solution:
    def pivotIndex(self, nums: List[int]) -> int:
        summary = sum(nums)
        left_summary = 0

        for i in range(len(nums)):
            if left_summary == summary - left_summary - nums[i]:
                return i
            left_summary += nums[i]

        return -1


# @lc code=end
